### PR TITLE
Discord Bot MVP

### PR DESCRIPTION
**Pull request by hammadj**
Setup of discord bot architecture

### CODE REVIEW

**Review by dysbulic — APPROVED** [review]
This seems pretty straightforward.

File: packages/discord-bot/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Use the Client that are provided by @typeit/discord NOT discord.js
+import { Client } from '@typeit/discord';
+
+import { CONFIG } from './config';
+
+async function start() {
+  const client = new Client({
+    classes: [
+      `${__dirname}/*Discord.ts`, // glob string to load the classes
```
Comment (dysbulic) [inline packages/discord-bot/src/index.ts:9]: This is causing the tests to fail.

**Comment by dysbulic** [comment]
The fixes to the build check (elements returning `undefined` which can't be serialized as JSON) are in #294.

File: packages/discord-bot/codegen.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+overwrite: true
+require:
+  - ts-node/register
+generates:
+  ./src/__generated__/hasura-sdk.ts:
+    schema: '../../schema.graphql'
+    documents:
+      - ./src/graphql/**/(!(*.d)).ts
```
Comment (dysbulic) [inline packages/discord-bot/codegen.yml:8]: This is causing the tests to fail when it doesn't match anything.

**Comment by pakokrew** [comment]
Maybe add dockerfile + compose configurations ?

**Comment by hammadj** [comment]
> The fixes to the build check (elements returning `undefined` which can't be serialized as JSON) are in #294.

Would be better to make changes directly to master next time. When PRs get too big and are trying to do too many things at once it doesn't work out well. If you want to split XP with another player you can write "paired with @username" in the PR description

**Comment by hammadj** [comment]
> Maybe add dockerfile + compose configurations ?

Im intending this to be used as a package / imported by the backend instead of running as its own service, so didn't make a Dockerfile for that reason. If you think its worth it just for dev purposes I can add it in

**Comment by pakokrew** [comment]
> > Maybe add dockerfile + compose configurations ?
> 
> Im intending this to be used as a package / imported by the backend instead of running as its own service, so didn't make a Dockerfile for that reason. If you think its worth it just for dev purposes I can add it in

Thought it would become a service of its own. Then I guess we don't need docker configs.

**Review by pakokrew — APPROVED** [review]

**Pull request merged**